Fix typos, add some punctuation, and rephrase some text (#28452)
akka/akka: akka-docs/src/main/paradox/typed/from-classic.md

Why is it called `Behavior` and not `Actor`?

In Typed, the `Behavior` defines how to handle incoming messages and after processing a message, a potentially

> different `Behavior` is returned for processing the next message. This means that an actor is started with an initial `Behavior`

Reviewer: I feel this changes the meaning somewhat, I'd want to change the `each` -> `a` but partially keep the next part: `may return a different behavior for processing the next message`

Author: @johanandren 

Would this be ok:

_In Typed, the `Behavior` defines how to handle incoming messages. After processing a message, a different `Behavior` may be returned for processing the next message._

Author: @johanandren Changed.

Corresponding method in Typed is called `spawn` in the @scala[`akka.actor.typed.scaladsl.ActorContext`]@java[`akka.actor.typed.javadsl.ActorContext`].

There is no `spawn` method in the @scala[`akka.actor.typed.scaladsl.ActorSystem`]@java[`akka.actor.typed.javadsl.ActorSystem`]

> for creating top level actors. Instead, there's a single top level actor defined by a user guardian `Behavior` that is given

Reviewer 2: ```suggestion
for creating top level actors. Instead, there is a single top level actor defined by a user guardian `Behavior` that is given
```

when starting the `ActorSystem`. Other actors are started as children of that user guardian actor or
children of other actors in the actor hierarchy. This is explained more in @ref:[ActorSystem](#actorsystem).

The `actorOf` method takes an `akka.actor.Props` parameter, which is like a factory for creating the actor instance, and it's
also used when creating a new instance when the actor is restarted. The `Props` may also define additional
properties such as which dispatcher to use for the actor.

Instead of creating an actor via a `Props` factory in classic, in typed, a `spawn` method creates an actor 

> from a given `Behavior`.

Reviewer: Since we have a typed props perhaps instead:

"In typed, the `spawn` method creates an actor directly from a given `Behavior` without using a `Props` factory. It does however accept an optional `akka.actor.typed.Props` for specifying Actor metadata."

Author: +1 for your alternative phrasing.

Author: I've added your suggested change.

when creating an `ActorSystem` in Typed you give it a `Behavior` that will be used as the top level actor, also known 
as the user guardian.

> Additional actors for an application can be created from the user guardian alongside performing the initialization

Reviewer: I think "can be created" sounds too optional, this changes the original meaning which is more of a "this is how you do it".

Author: So instead of `Additional actors for an application can be created`, you would change this to `Additional actors for an application are created`, correct?

Reviewer: That is better 👍

Author: Done.

## Supervision

> An important difference between classic and typed is that, in typed, actors are by default stopped if

Reviewer 3: ```suggestion
An important difference between classic and typed is that in typed, actors are stopped by default if
```

an exception is thrown and no supervision strategy is defined. In contrast,

> in Classic, by default, actors are restarted.

Reviewer 3: ```suggestion
in classic, by default, actors are restarted.
```

* @ref:[Classic](../fault-tolerance.md)
* @ref:[Typed](fault-tolerance.md)

> ## Lifecycle hooks

Reviewer 2: Oops ;)

children. Therefore, this is not a useful way to lookup children when the purpose is to send
messages to them.

> Instead of finding children via the `ActorContext`, it's recommended to use an application specific

Reviewer 2: ```suggestion
Instead of finding children via the `ActorContext`, it is recommended to use an application specific
```
(not your text, but this reads better to me ;) )

Starting an actor on a remote node—so called remote deployment—isn't supported in Typed.

The main reason is that we would discourage this feature since it often results in a too tight coupling

> between nodes and failure handling. For example if the node of the parent actor crashes,

Reviewer 3: ```suggestion
This feature would be discouraged because it often results in tight coupling between nodes and undesirable failure handling. For example if the parent actor's node crashes,
```

Author: I new understood what _**undesirable** failure handling_ means... I guess it's the _handling_ that's undesirable, but I read it as _undesirable failure_.
Can you clarify?

Author: I added your suggestion in [317dd1c](https://github.com/akka/akka/pull/28452/commits/317dd1c4432287f02e242947534c9c8cc5eba4ee)

## Synchronous Testing

> Classic and typed have different Test Kits.

Reviewer: This changes the meaning, I think the intention was to convey that the design of the synchronous testkit APIs are very different (TestActor vs BehaviorTestKit), this just reads as that they are two different test kits.

Author: How about: `Classic and typed have completely different Test Kits`?

Reviewer: The complication is that this is just about the synchronous testing, the async testing is more similar between the two.

Author: or `Classic and typed have completely different Test Kits for synchronous testing`

Reviewer: Sounds good to me :)

Author: I guess this is close to the original. What was missing there what the different _between x and y_ part

Author: Added my latest suggestion.

> The Test Kits for synchronous testing are completely different between classic and typed.

Reviewer 3: Or is it clearer to say
```suggestion
Classic and typed have different Test Kits.
```
